RELEASES.md

```diff
@@ -1,3 +1,119 @@
+Version 1.18.0 (2017-06-08)
+===========================
+
+Language
+--------
+
+- [Stabilize pub(restricted)][40556] `pub` can now accept a module path to
+  make the item visible to just that module tree. Also accepts the keyword
+  `crate` to make something public to the whole crate but not users of the
+  library. Example: `pub(crate) mod utils;`.
+- [Stabilize `#![windows_subsystem]` attribute][40870] conservative exposure of the
+  `/SUBSYSTEM` linker flag on Windows platforms.
+- [Refactor of trait object type parsing][40043] Now `ty` in macros can accept
+  types like `Write + Send`, trailing `+` are now supported in trait objects,
+  and better error reporting for trait objects starting with `?Sized`.
+- [0e+10 is now a valid floating point literal][40589]
+- [Now warns if you bind a lifetime parameter to 'static][40734]
+
+Compiler
+--------
+
+- [LLVM backend upgraded to 4.0][40123]
+- [rustc can now emit mir with `--emit mir`][39891]
+- [Improved LLVM IR for trivial functions][40367]
+- [Added explanation for E0090(Wrong number of lifetimes are supplied)][40723]
+- [Rustc is now faster][41469]
```

Comment:
According to [performance graph](http://perf.rust-lang.org/graphs.html?start=2017-04-12T09%3A16%3A14%2B00%3A00&end=2017-05-12T07%3A33%3A10%2B00%3A00&crates=futures-rs-test-all%2Chelloworld%2Chtml5ever-2016-08-25%2Chyper.0.5.0%2Cinflate-0.1.0%2Cissue-32062-equality-relations-complexity%2Cissue-32278-big-array-of-strings%2Cjld-day15-parser%2Cpiston-image-0.10.3%2Cregex-0.1.80%40010-baseline%2Cregex-0.1.80%40020-incr-from-scratch%2Cregex-0.1.80%40030-compile_one%2Cregex-0.1.80%40040-is_valid_cap_letter%2Cregex-0.1.80%40050-expand%2Cregex-0.1.80%40060-Compiler-new%2Cregex-0.1.80%40070-BYTE_FREQUENCIES%2Cregex-0.1.80%40080-SparseSet%2Cregex-0.1.80%40090-Job%2Cregex-0.1.80%40100-incr-no-change%2Cregex.0.1.30%2Crust-encoding-0.3.0%2Csyntex-0.42.2%40000-base%2Csyntex-0.42.2%40010-incr%2Csyntex-0.42.2%40020-clean%2Ctuple-stress&phases=AST%20validation%2CLLVM%20passes%2CMIR%20cleanup%20and%20validation%2CMIR%20dump%2CMIR%20optimisations%2Cassert%20dep%20graph%2Cattribute%20checking%2Cborrow%20checking%2Cchecking%20for%20inline%20asm%20in%20case%20the%20target%20doesn%27t%20support%20it%2Ccoherence%20checking%2Ccomplete%20gated%20feature%20checking%2Ccompute_incremental_hashes_map%2Cconst%20checking%2Ccrate%20injection%2Cdeath%20checking%2Cdrop-impl%20checking%2Cearly%20lint%20checks%2Ceffect%20checking%2Cexpansion%2Cimpl%20wf%20inference%2Cindexing%20hir%2Cintrinsic%20checking%2Citem-bodies%20checking%2Citem-types%20checking%2Clanguage%20item%20collection%2Clifetime%20resolution%2Clinking%2Clint%20checking%2Cliveness%20checking%2Cload_dep_graph%2Clooking%20for%20entry%20point%2Clooking%20for%20plugin%20registrar%2Cloop%20checking%2Clowering%20ast%20-%3E%20hir%2Cmatch%20checking%2Cmaybe%20building%20test%20harness%2Cmaybe%20creating%20a%20macro%20crate%2Cname%20resolution%2Cparsing%2Cplugin%20loading%2Cplugin%20registration%2Cprivacy%20checking%2Creachability%20checking%2Crecursion%20limit%2Cregion%20resolution%2Cresolving%20dependency%20formats%2Crvalue%20checking%2Cserialize%20dep%20graph%2Cserialize%20work%20products%2Cstability%20checking%2Cstability%20index%2Cstatic%20item%20recursion%20checking%2Ctranslation%2Ctype%20collecting%2Cunused%20lib%20feature%20checking%2Cvariance%20inference%2Cvariance%20testing%2Cwf%20checking&group_by=crate), the compilation time was reduced around 15% to 20% across the board if I am not mistaken. This is pretty significant improvement. We might as well let user know the actual number.